openmls/src/group/core_group/validation.rs

```diff
@@ -264,4 +266,81 @@ impl CoreGroup {
 
         Ok(())
     }
+
+    /// Validate constraints on an external commit. This function implements the following checks:
+    ///  - ValSem240: External Commit, inline Proposals: There MUST be at least one ExternalInit proposal.
+    ///  - ValSem241: External Commit, inline Proposals: There MUST be at most one ExternalInit proposal.
+    ///  - ValSem242: External Commit, inline Proposals: There MUST NOT be any Add proposals.
+    ///  - ValSem243: External Commit, inline Proposals: There MUST NOT be any Update proposals.
+    ///  - ValSem244: External Commit, inline Remove Proposal: The identity and the endpoint_id of the removed
+    ///               leaf are identical to the ones in the path KeyPackage.
+    ///  - ValSem245: External Commit, referenced Proposals: There MUST NOT be any ExternalInit proposals.
+    pub(crate) fn validate_external_commit(
+        &self,
+        proposal_queue: &ProposalQueue,
+        path_key_package_option: Option<&KeyPackage>,
+    ) -> Result<(), CoreGroupError> {
+        let mut external_init_proposals =
+            proposal_queue.filtered_by_type(ProposalType::ExternalInit);
+        // ValSem240: External Commit, inline Proposals: There MUST be at least one ExternalInit proposal.
+        if let Some(external_init_proposal) = external_init_proposals.next() {
+            // ValSem245: External Commit, referenced Proposals: There MUST NOT be any ExternalInit proposals.
+            if external_init_proposal.proposal_or_ref_type() == ProposalOrRefType::Reference {
+                return Err(ExternalCommitValidationError::ReferencedExternalInitProposal.into());
+            }
+        } else {
+            return Err(ExternalCommitValidationError::NoExternalInitProposals.into());
+        };
+
+        // ValSem241: External Commit, inline Proposals: There MUST be at most one ExternalInit proposal.
+        if let Some(additional_external_init_proposal) = external_init_proposals.next() {
+            // ValSem245: External Commit, referenced Proposals: There MUST NOT be any ExternalInit proposals.
+            if additional_external_init_proposal.proposal_or_ref_type()
```

Comment:
Not sure we need this error, because we checked for that before. This check should just be about the fact that there is more than one ExternalInit.

Reply:
Yeah, I thought it would be good to use the error to indicate what exactly went wrong. But I guess it doesn't really matter.

The check above only checks for the first ExternalInit proposal. This one checks for the second. But I agree, that we can simply return the second error here, i.e. `ExternalCommitValidationError::MultipleExternalInitProposals`.